client/src/index.ts

```diff
@@ -29,8 +25,7 @@ export abstract class Repository<T extends XataRecord> extends Query<T> {
 
   abstract delete(id: string): void;
 
-  // Used by the Query object internally
-  abstract _runQuery<R extends XataRecord, Options extends QueryOptions<T>>(
+  abstract query<R extends XataRecord, Options extends QueryOptions<T>>(
```

Comment:
Since the parameters are public, expose ``query`` method instead of making it internal or protected. It's a ``READ`` operation in CRUD.